lib/CrashHandler/CrashHandler.cpp

```diff
@@ -430,21 +473,25 @@ void logBacktrace() try {
             boost::core::scoped_demangled_name demangled(&mangled[0]);
 
             if (demangled.get()) {
-              appendNullTerminatedString(demangled.get(), p);
+              appendNullTerminatedString(demangled.get(), BYTES_LEFT(buffer, p),
+                                         p);
             } else {
               // demangling has failed.
               // in this case, append mangled version. still better than nothing
-              appendNullTerminatedString(mangled, p);
+              appendNullTerminatedString(mangled, BYTES_LEFT(buffer, p), p);
             }
             // print offset into function
-            appendNullTerminatedString(" (+0x", p);
-            appendHexValue(reinterpret_cast<unsigned char const*>(&offset),
-                           sizeof(decltype(offset)), p, true);
-            appendNullTerminatedString(")", p);
+            if (BYTES_LEFT(buffer, p) >= 20) {
```

Comment:
appendHexValue will append 16 bytes, and the string before can append 5 bytes (` (+0x`).
```suggestion
            if (BYTES_LEFT(buffer, p) >= 21) {
```